### PR TITLE
feat: PR review environment

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Creates a PR review environment using Lambda and function URLs.
Each PR against the `main` branch will create a unique review
environment hooked up to the Staging API.

# Related
* cds-snc/notification-planning#625

### CODE REVIEW

**Review by jzbahrai — APPROVED** [review]

**Comment by sastels** [comment]
The only thing is that we do use redis in admin, for the initial sign on flow and for the template reviews. Is there any way to hook this up to redis somewhere?

**Review by sastels — APPROVED** [review]
So awesome!!!!

File: .github/workflows/test-admin-remove.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Remove test admin deployment
+
+on:
+  pull_request:
+    branches:
+      - main  
+    types: [closed]
```
Comment (jimleroyer) [inline .github/workflows/test-admin-remove.yaml:7]: That's very handy, to remove a preview app on closed PR. 

We got a few Snyk PR getting opened though and I wonder how many images/lambdas we'll have deployed, considering we are not disciplined at the moment (we plan to change that). Could we apply a filter? or a time based deletion of the lambdas? (which then might require a manual run of the workflow action if we want to spin one up again).

**Review by jimleroyer — COMMENTED** [review]
Nicely done @patheard

File: .github/workflows/test-admin-remove.yaml (hunk 1 above)
Comment (patheard) [reply .github/workflows/test-admin-remove.yaml:7]: Yeah, I've been thinking about that idea that the environments are deleted after a certain amount of time and if you need them back, the workflow can just be manually re-run.

Comment (patheard) [reply .github/workflows/test-admin-remove.yaml:7]: There, added this feature in #1274

**Pull request merged**